src/utils/grid_rendering.py

```diff
@@ -0,0 +1,79 @@
+from model.grid import Grid
+from model.point import Point
+from PIL import Image, ImageDraw, ImageFont
+
+
+def display_grid(grid: Grid):
+    """
+    Display the grid in a human-readable format directly to standard output.
+    """
+    for row_index in range(1, 10):
+        if row_index % 3 == 1:
+            print("+" + "-" * 7 + "+" + "-" * 7 + "+" + "-" * 7 + "+")
+
+        for col_index in range(1, 10):
+            if col_index % 3 == 1:
+                print("|", end=" ")
+
+            cell = grid[Point(col_index - 1, row_index - 1)]
+            if cell is None or cell.value is None:
+                print(".", end=" ")
+            else:
+                print(cell.value, end=" ")
+
+        print("|")
+
+    print("+" + "-" * 7 + "+" + "-" * 7 + "+" + "-" * 7 + "+")
+
+
+def render_grid(grid: Grid) -> Image.Image:
```

Comment:
Rendering the grid as a graphic makes it pleasant on the eye for reviewing the state.
This might be useful to generate a walkthrough of the solving steps in a report when all the solvers are implemented.